tests/Functional/Execution/AbstractTest.php

```diff
@@ -14,31 +13,161 @@
 use function Digia\GraphQL\Type\GraphQLObjectType;
 use function Digia\GraphQL\Type\GraphQLSchema;
 use function Digia\GraphQL\Type\GraphQLString;
+use function Digia\GraphQL\Type\GraphQLUnionType;
 
 class AbstractTest extends TestCase
 {
     // EXECUTE: HANDLES EXECUTION OF ABSTRACT TYPES
 
     /**
      * isTypeOf used to resolve runtime type for Interface
+     *
      * @throws \Digia\GraphQL\Error\InvariantException
      * @throws \Digia\GraphQL\Error\SyntaxErrorException
      */
-    public function testResolveFunctionForInterface()
+    public function testIsTypeOfUsedToResolveFunctionForInterface()
     {
+        $PetInterfaceType = GraphQLInterfaceType([
+            'name'   => 'Pet',
+            'fields' => [
+                'name' => ['type' => GraphQLString()]
+            ]
+        ]);
+
+        $DogType = GraphQLObjectType([
```

Comment:
Not really the same object, basically we can't use Dog() for this, because it will create two identical objects in the schema. That's yesterday issue with `===`